File: Greencoin/GreenCoinMobile/lib/models/Business.dart
import 'Address.dart';
import 'Wallet.dart';

class Business {
  String id;
  String firebaseUid;
  String municipalityId;
  Address address;
  Wallet wallet;
  String name;
  String phoneNumber;
  String ownerFirstName;
  String ownerLastName;
  String category;
  String about;
  List<String> imagesUris;

  Business(
      {this.id,
      this.firebaseUid,
      this.municipalityId,
      this.address,
      this.wallet,
      this.name,
      this.phoneNumber,
      this.ownerFirstName,
      this.ownerLastName,
      this.category,
      this.about,
      this.imagesUris});

  Business.fromJson(Map<String, dynamic> json) {
    id = json['Id'];
    firebaseUid = json['FirebaseUid'] != null ? json['FirebaseUid'] : null;
    municipalityId = json['MunicipalityId'];
    address =
        json['Address'] != null ? new Address.fromJson(json['Address']) : null;
    wallet =
        json['Wallet'] != null ? new Wallet.fromJson(json['Wallet']) : null;
    name = json['Name'];
    phoneNumber = json['PhoneNumber'];
    ownerFirstName = json['OwnerFirstName'];
    ownerLastName = json['OwnerLastName'];
    category = json['Category'];
    about = json['About'];
    imagesUris =
        json['ImagesUris'] != null ? json['ImagesUris'].cast<String>() : [];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    if (this.id != null) data['Id'] = this.id;
    data['FirebaseUid'] = this.firebaseUid;
    data['MunicipalityId'] = this.municipalityId;
    if (this.address != null) {
      data['Address'] = this.address.toJson();
    }
    if (this.wallet != null) {
      data['Wallet'] = this.wallet.toJson();
    }
    data['Name'] = this.name;
    data['PhoneNumber'] = this.phoneNumber;
    data['OwnerFirstName'] = this.ownerFirstName;
    data['OwnerLastName'] = this.ownerLastName;
    data['Category'] = this.category;
    data['About'] = this.about;
    if (this.imagesUris != null) data['ImagesUris'] = this.imagesUris;
    return data;
  }
}

File: Greencoin/GreenCoinWebsite/src/Components/Business.js
import React,{useEffect, useState} from 'react'
import {MDBCard, MDBCardBody, MDBCardImage,MDBBtn ,MDBModal, MDBModalBody, MDBModalHeader,MDBModalFooter} from
'mdbreact';
import {baseUrl} from '../../package.json'


import { MDBDataTable } from 'mdbreact';
import axios from 'axios';
import BussInfo from './BussInfo';

export default function Business() 
{
  const [modal, setmodal] = useState(false)
  const [id, setid] = useState('')
  let content;
  const [business, setbusiness] = useState(['']);
  const [loading, setloading] = useState(true);
  const url=baseUrl+'/api/Businesses?municipalityId=0713b854-89ef-497b-a61c-e0bdb669fa70'
  useEffect(
    () => {
      
      axios.get(url).then(
            res => {
              setbusiness(res.data.Businesses);
              console.log(business);
              setloading(false);
                  }
     )  

},[url]);
const toggle =() =>{
  setmodal(!modal)
}

 const handleClick = (d) =>{
          setmodal(true)    
          setid(d)
                     
  console.log(d)
}

     if(!loading)
     {
     
            content=( business.map(
                  d =>(
                    {
                      id:d.Id,
                      name:d.Name,
                      owner:d.OwnerFirstName +' '+d.OwnerLastName,
                      category:d.Category,
                      phone:d.PhoneNumber,
                      clickEvent: () => {
                        handleClick(d.Id)
                      } 
                    } 
                      )
                )
            ) 
     }
      
     
   
   const  dataList = {
        columns: [
          {
            label:'ID',
            field:'id',
            sort :'asc',
            width:100,
           
          },
          
           {
            label: 'Name',
            field: 'name',
            sort: 'asc',
            width: 150
          },
          {
            label: 'Owner Name',
            field: 'owner',
            sort: 'asc',
            width: 150
          },
          {
            label: 'Category',
            field: 'category',
            sort: 'asc',
            width: 270
          },
          {
            label: 'Phone Number',
            field: 'phone',
            sort: 'asc',
            width: 200
          }
          
          
        ],
       
      
        rows: content
      };
      
    

    return (
      
        <div>
         
        <MDBCard>
          <MDBCardImage
            className='narrow-table primary-color white-text' tag='div' >
            <h2 className='h2-responsive'>Business List </h2>
          </MDBCardImage>
          <MDBCardBody >
           {!loading ?  <MDBDataTable small   data={dataList}/> :<div className="spinner-grow text-success" role="status">
      <span className="sr-only">Loading...</span>
    </div>} 
          </MDBCardBody>
        </MDBCard>

          {modal && <MDBModal isOpen={modal} toggle={toggle}>
                             <MDBModalHeader toggle={toggle}>{id}</MDBModalHeader>
                               <MDBModalBody>
                              <BussInfo ID={id}/>
                               </MDBModalBody>
                               <MDBModalFooter>
                                 <MDBBtn color="primary" onClick={toggle}>Close</MDBBtn>
                                 </MDBModalFooter>
                        </MDBModal>}
        </div>

      
        
    
    )
   
}
